src/Overlay.tsx

```diff
@@ -181,18 +181,20 @@ const Overlay = React.forwardRef<HTMLElement, OverlayProps>(
       >
         {(overlayProps, { arrowProps, placement, popper: popperObj, show }) => {
           wrapRefs(overlayProps, arrowProps);
+          // Need to get placement from popper object first, handling case when overlay is flipped using 'flip' prop
+          const updatedPlacement = popperObj?.placement || placement;
```

Comment:
Hmm I wonder if we need to fallback to `placement` here.

I think(?) popper's placement should always be defined.

I think this might be enough `const updatedPlacement = popperObj?.placement;`

Reply:
i think you're right, will change